### ISSUE TITLE
Avoid adding hooks in class constructors

### ISSUE THREAD

**Issue by jdevalk** [equalizedigital/accessibility-checker#395]
Ideally, hooks should not run in an object's constructor. The constructor should be used to initialize the object, not to run hooks. Hooks should be run from methods. This is not a hard rule, but it's a good practice to follow.

For example, the `\EDAC\Admin_Notices` class (and others): we should move the hooks to a separate method (like `init_hooks`), and then call that method when needed.  

```php
$admin_notices = new \EDAC\Admin_Notices();
// Added this line for the sake of this example.
$admin_notices->init_hooks();
$notice_text = $admin_notices->edac_password_protected_notice_text();
```

This way, we can easily get a different instance of an object when needed without adding the same hooks again and again. In the case of `Admin_Notices` we only need a single instance of the object, but that is not always the case.

Especially when we start adding unit tests to the code, we'll need to fix this everywhere.
